src/cmd/cli/command/deploymentinfo_test.go

```diff
@@ -44,80 +44,102 @@ func TestPrintServiceStatesAndEndpoints(t *testing.T) {
 
-	_ = printServiceStatesAndEndpoints([]*defangv1.ServiceInfo{
+	tests := []struct {
+		name          string
+		serviceinfos  []*defangv1.ServiceInfo
+		expectedLines []string
+	}{
 		{
-			Service: &defangv1.Service{
-				Name: "service1",
-				Ports: []*defangv1.Port{
-					{Mode: defangv1.Mode_INGRESS},
-					{Mode: defangv1.Mode_HOST},
+			name: "empty endpoint list",
+			serviceinfos: []*defangv1.ServiceInfo{
+				{
+					Service: &defangv1.Service{
+						Name: "service1",
+						Ports: []*defangv1.Port{
+							{Mode: defangv1.Mode_INGRESS},
+							{Mode: defangv1.Mode_HOST},
+						},
+					},
+					Status:     "UNKNOWN",
+					Domainname: "example.com",
+					Endpoints:  []string{},
 				},
 			},
-			Status: "UNKNOWN",
-			Endpoints: []string{
-				"example.com",
-				"service1.internal",
+			expectedLines: []string{
+				"Deployment  Name      Status         Endpoints  DomainName",
+				"            service1  NOT_SPECIFIED  N/A        https://example.com",
+				" * Run `defang cert generate` to get a TLS certificate for your service(s)",
+				"",
 			},
-		}})
-	const expectedOutput = `Deployment  Name      Status         Endpoints
-            service1  NOT_SPECIFIED  example.com, service1.internal
-`
-	receivedLines := strings.Split(stdout.String(), "\n")
-	expectedLines := strings.Split(expectedOutput, "\n")
-
-	if len(receivedLines) != len(expectedLines) {
-		t.Errorf("Expected %v lines, received %v", len(expectedLines), len(receivedLines))
-	}
-
-	for i, receivedLine := range receivedLines {
-		receivedLine = strings.TrimRight(receivedLine, " ")
-		if receivedLine != expectedLines[i] {
-			t.Errorf("\n-%v\n+%v", expectedLines[i], receivedLine)
-		}
-	}
-}
-
-func TestPrintServiceStatesAndEndpointsAndDomainname(t *testing.T) {
-	defaultTerm := term.DefaultTerm
-	t.Cleanup(func() {
-		term.DefaultTerm = defaultTerm
-	})
-
-	var stdout, stderr bytes.Buffer
-	term.DefaultTerm = term.NewTerm(os.Stdin, &stdout, &stderr)
-
-	_ = printServiceStatesAndEndpoints([]*defangv1.ServiceInfo{
+		},
 		{
-			Service: &defangv1.Service{
-				Name: "service1",
-				Ports: []*defangv1.Port{
-					{Mode: defangv1.Mode_INGRESS},
-					{Mode: defangv1.Mode_HOST},
+			name: "Service with Domainname",
+			serviceinfos: []*defangv1.ServiceInfo{
+				{
+					Service: &defangv1.Service{
+						Name: "service1",
+						Ports: []*defangv1.Port{
+							{Mode: defangv1.Mode_INGRESS},
+							{Mode: defangv1.Mode_HOST},
+						},
+					},
+					Status:     "UNKNOWN",
+					Domainname: "example.com",
+					Endpoints: []string{
+						"example.com",
+						"service1.internal:80",
+					},
 				},
 			},
-			Status:     "UNKNOWN",
-			Domainname: "example.com",
-			Endpoints: []string{
-				"example.com",
-				"service1.internal",
+			expectedLines: []string{
+				"Deployment  Name      Status         Endpoints                                  DomainName",
+				"            service1  NOT_SPECIFIED  https://example.com, service1.internal:80  https://example.com",
```

Comment:
this can't be be right, HTTPS to 80?

Reply:
it's not https to 80. it's https to example.com.